_cite/util.py

```diff
@@ -49,18 +49,35 @@ def log(message="\n--------------------\n", indent=0, level="", newline=True):
         "SUCCESS": "[black on #10B981]",
         "INFO": "[grey70]",
     }
-    color = palette.get(level, "") or palette.get(indent, "") or "[white]"
+    color = get_safe(palette, level, "") or get_safe(palette, indent, "") or "[white]"
     if newline:
         print()
     print(indent * "    " + color + str(message) + "[/]", end="", flush=True)
 
 
 def label(entry):
     """
-    get "label" of dict entry
+    get "label" of dict entry (for logging purposes)
     """
 
-    return list(entry.keys())[0] + ": " + list(entry.values())[0]
+    return str(list(entry.keys())[0]) + ": " + str(list(entry.values())[0])
+
+
+def get_safe(item, path, default=None):
```

Comment:
So I guess this would sort of be a replacement for safe_get? Does it also work with lists? Does it also work safely if the root object (the value you're passing) is falsey or otherwise not a list/dict, because that's something the safe_get does, which was actually the cause of that orcid bug.

Reply:
`Box` is interesting, but it's more dependency and you'd have to wrap each of the elements you want to dereference in a `Box()` constructor. That would run against what this method is for, which AFAIK is for saving typing over chaining calls to `.get()` with a safe default for further `get()` calls. The `Box()` constructor also doesn't accept `None`, falsey, or even scalar values; it only works with mappings/iterables (i.e., lists and dicts).